bcgov/cas-pipeline
# chore/57 tf bootstrap migration

Addresses [ZH#57](https://github.com/bcgov/cas-reporting/issues/57).
Moves general Terraform bootstrapping scripts from testing in Cas-Registration to here and then applying them to all CAS projects. 

## Changes 🚧

- Added script to `make provision` target that creates storage buckets for all CAS projects. Like other parts of provisioning, this has been made idempotent.
- New variable `GCP_PROJECT` added to `.env-example` (updated in shared password manager  🔑)
- New variable object `gcpTerraform` added to `helm/cas-provision/values.yaml`  (updated in shared password manager 🔑)
 - Added Helm template to provision secrets related to Terraform in OpenShift `gcp-credentials-secret`
   - `gcp-project-id`: Google Cloud Platform project ID where storage buckets for state are created
   - `tf_backend`: backend file for Terraform
   - `sa_json`: service account credentials

## To test 🔬

- Use the GCP credentials from the password manager and authenticate the service account with `gcloud auth activate-service-account SERVICE_ACCOUNT@DOMAIN.COM --key-file=/path/gcp-tf-credentials.json --project=$GCP_PROJECT`. (see the note in the password manage for the full command). 
- With the updated .env and values files, you can run `make provision`. 

## Notes📝 
- The shared env and values files have been updated in the shared CAS password manager. 
- While the script to create storage buckets using `gcloud cli` is idempotent, `gcloud` throws a warning when storage buckets are already created.
